documentation/new-version-guideline.md

```diff
@@ -0,0 +1,302 @@
+        ClientOptions: policy.ClientOptions{
+            Retry: policy.RetryOptions{
+                // retry for 5 times
+                MaxRetries: 5,
+            },
+        },
+    },
+)
+```
+
+### Customized Policy
+
+You can use `arm.ClientOptions.PerCallPolicies` and `arm.ClientOptions.PerRetryPolicies` option to inject customized policy to the pipeline. You can refer to `azcore` [document](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore) for further information.
+
+### Custom HTTP Client
+
+You can use `arm.ClientOptions.Transport` to set your own implementation of HTTP client. The HTTP client must implement the `policy.Transporter` interface. For example:
+
+```go
+// your own implementation of HTTP client
+httpClient := NewYourOwnHTTPClient{}
+rgClient, err := armresources.NewResourceGroupsClient(subscriptionId, credential,
+    &arm.ClientOptions{
+        ClientOptions: policy.ClientOptions{
+            Transport: &httpClient,
+        },
+    },
+)
+```
+
+### Reference
+
+More client options can be found [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/policy/policy.go).
+
+
+## Troubleshooting
+
+### Logging
+
+The SDK uses the classification-based logging implementation in `azcore`. To enable console logging for all SDK modules, set AZURE_SDK_GO_LOGGING to all. 
+
+Use `LogOption` to configure log behavior. For example:
+```go
+rgClient, err := armresources.NewResourceGroupsClient(subscriptionId, credential,
+    &arm.ClientOptions{
+        ClientOptions: policy.ClientOptions{
+            Logging: policy.LogOptions{
+                // include HTTP body for log
+                IncludeBody: true,
+            },
+        },
+    },
+)
+```
+
+Use the `azcore/log` package to control log event and write log to the desired location. For example:
+
+```go
+import azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
+
+// print log output to stdout
+azlog.SetListener(func(event azlog.Event, s string) {
+    fmt.Println(s)
+})
+
+// include only azidentity credential logs
+azlog.SetEvents(azidentity.EventAuthentication)
+```
+
+### Raw HTTP response
+When there is an error in the SDK request, you need to convert the error to the `azcore.ResponseError` interface to get the raw HTTP response. When the request is successful, you can get the raw HTTP response from request context.
+
+```go
+var rawResponse *http.Response
+ctxWithResp := runtime.WithCaptureResponse(context.TODO(), &rawResponse)
+resp, err := resourceGroupsClient.CreateOrUpdate(ctxWithResp, resourceGroupName, resourceGroupParameters, nil)
+if err != nil {
+    var respErr *azcore.ResponseError
+    if errors.As(err, &respErr) {
+        log.Fatalf("Status code: %d", respErr.RawResponse.StatusCode)
+    } else {
+        log.Fatalf("Other error: %+v", err)
+    }
+}
+log.Printf("Status code: %d", rawResponse.StatusCode)
+```
+
+> NOTE: runtime is import from **github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime**
```

Comment:
Could we add the import clause to the code snippet? and maybe with an alias `armruntime` to disambiguate